type_check_decorator.py

```diff
@@ -0,0 +1,74 @@
+from collections.abc import Callable
+from functools import wraps
+from typing import Any, Type
+
+
+class InvalidTypeError(Exception):
+    
+    """
+    A InvalidTypeError class, inherits from Exception.
+    A custom exception that receives correct type and an object and print what the type should be.
+    :ivar correct_type : The type that should be received.
+    :ivar receive_arg : An argument that have been received.
+
+    :param correct_type : The type that should be received.
+    :param receive_arg : An argument that have been received.
+    """
+    def __init__(self, correct_type: Type[object], receive_arg: Any):
+        self.correct_type = correct_type
+        self.receive_arg = receive_arg
+
+    def __str__(self):
+        return f"You have been entered '{self.receive_arg}' which from type {type(self.receive_arg)}."\
+               f" You should enter {self.correct_type} type"
+
+
+def type_check(correct_type: Type[object]) -> Callable[Callable[Any, Any], Callable[Any, Any]]:
+    """
+    Returns decorator that checks if the given type is like the function's argument type.
+
+    :param correct_type: The type that should send to function.
+    :return: Decorator that will check if the type like wanted and will raise an error if it isn't.
+    """
+    @wraps(correct_type)
+    def type_check_decorator(function: Callable[correct_type, Any]) -> Callable[correct_type, Any]:
+        """
+        Checking if a function that should receive one argument will receive the correct type of argument
+        If the argument type isn't as wanted, InvalidTypeError will raise.
+        :param function: Function that receive one argument.
```

Comment:
A* function that

Reply:
You want your documentation to be valid English sentences so you'll be able to create a formal documentation from them using an external tool